opentelemetry-otlp/src/exporter/http/trace.rs

```diff
@@ -46,8 +46,24 @@ impl SpanExporter for OtlpHttpClient {
         }
 
         Box::pin(async move {
+            #[cfg(not(debug_assertions))]
             client.send(request).await?;
 
+            #[cfg(debug_assertions)]
+            {
+                let request_uri = request.uri().to_string();
+                let response = client.send(request).await?;
+
+                if !response.status().is_success() {
+                    let error = format!(
+                        "OpenTelemetry export failed. Url: {}, Response: {:?}",
+                        request_uri,
+                        response.body()
```

Comment:
can we just log the statuscode alone and avoid the debugonly part?

Reply:
The response often tells you whats _actually_ wrong with your setup.
The debugonly part is added to avoid allocating `request_uri` - this was the first issue in my case - I did not have the `OTEL_EXPORTER_OTLP_TRACES_ENDPOINT` variable set and all requests were going to `https://api.honeycomb.io/`. A `404` status could would not have made it any better.

Reply:
do we really need to worry about the allocation issue? It should only occur *if* there is a failure, and also only once every few seconds (export interval).

Reply:
> The debugonly part is added to avoid allocating request_uri 

Just wondering if we need to log request_uri, as the collector URL is a known entity configured during bootstrap.

Reply:
> and also only once every few seconds (export interval).

Yeah it's probably not an issue after all 😄 .  I still prefer logging the request URI because that was the first thing I was struggling to set correctly

> this was the first issue in my case - I did not have the OTEL_EXPORTER_OTLP_TRACES_ENDPOINT variable set and all requests were going to https://api.honeycomb.io/. A 404 status could would not have made it any better.

Here is the commit: https://github.com/open-telemetry/opentelemetry-rust/pull/1498/commits/5fd5a9f04db5830a14f2fa8f46e1492650ef7586, thanks for your feedback.